gate failure: running `make import-boss`

<!--
Please add the affected binary name in the title unless multiple binaries are affected, e.g.
[cinder-csi-plugin] Cannot delete PV
For openstack-cloud-controller-manager, you can use [occm] for short.

All the currently maintained binaries are:
* openstack-cloud-controller-manager (occm)
* cinder-csi-plugin
* manila-csi-plugin
* k8s-keystone-auth
* client-keystone-auth
* octavia-ingress-controller
* magnum-auto-healer
* barbican-kms-plugin
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
```
[root@m5404019 cloud-provider-openstack]# make import-boss
echo "installing import-boss"
installing import-boss
go get -u k8s.io/code-generator/cmd/import-boss
go: downloading k8s.io/code-generator v0.21.0
go: found k8s.io/code-generator/cmd/import-boss in k8s.io/code-generator v0.21.2
go: downloading k8s.io/klog/v2 v2.9.0
go: downloading k8s.io/gengo v0.0.0-20210707163844-888f71863374
go: downloading github.com/go-logr/logr v1.0.0
go: downloading golang.org/x/tools v0.1.4
go: downloading golang.org/x/sys v0.0.0-20210630005230-0f9fa26af87c
go: downloading golang.org/x/mod v0.4.2
go: gopkg.in/yaml.v2 upgrade => v2.4.0
go: sigs.k8s.io/yaml upgrade => v1.2.0
go: golang.org/x/xerrors upgrade => v0.0.0-20200804184101-5ec99f83aff1
go: golang.org/x/sys upgrade => v0.0.0-20210630005230-0f9fa26af87c
go: k8s.io/klog/v2 upgrade => v2.9.0
go: github.com/spf13/pflag upgrade => v1.0.5
go: golang.org/x/tools upgrade => v0.1.4
go: golang.org/x/mod upgrade => v0.4.2
go: github.com/go-logr/logr upgrade => v1.0.0
go: k8s.io/gengo upgrade => v0.0.0-20210707163844-888f71863374
# k8s.io/klog/v2
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:705:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:724:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:742:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:763:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:782:11: invalid operation: loggr != nil (mismatched types logr.Logger and nil)
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:783:3: undefined: logr.WithCallDepth
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:794:11: invalid operation: loggr != nil (mismatched types logr.Logger and nil)
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:795:3: undefined: logr.WithCallDepth
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:915:9: invalid operation: log != nil (mismatched types logr.Logger and nil)
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:919:4: undefined: logr.WithCallDepth
.go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:919:4: too many errors
make: *** [Makefile:186: import-boss] Error 2

```

looks like someone has same issue
https://golangrepo.com/repo/go-logr-logr

not sure we need some changes or rely on other repo to act, tracking here

**What you expected to happen**:


**How to reproduce it**:


**Anything else we need to know?**:


**Environment**:
- openstack-cloud-controller-manager(or other related binary) version:
- OpenStack version:
- Others:

--- a/Makefile
+++ b/Makefile
@@ -183,7 +183,7 @@ endif
 import-boss:
 ifndef HAS_IMPORT_BOSS
 	echo "installing import-boss"
-	go get -u k8s.io/code-generator/cmd/import-boss
+	go install k8s.io/code-generator/cmd/import-boss
 endif
 	hack/verify-import-boss.sh
 

